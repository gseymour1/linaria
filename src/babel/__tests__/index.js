/* @flow */

const babel = require('@babel/core');
const dedent = require('dedent');

it('should extract CSS to a comment', async () => {
  const { code } = await babel.transformAsync(
    dedent`
    var background = 'yellow';

    const Container = styled('div')\`
      font-family: ${'${serif}'};
      background-color: ${'${background}'};
      color: ${'${props => props.color}'};
      width: ${'${100 / 3}'}%;
      border: 1px solid red;

      &:hover {
        border-color: blue;
      }
    \`;
    `,
    {
      plugins: [require.resolve('../index')],
      filename: '/app/index.js',
    }
  );

  expect(code).toMatchSnapshot();
});
