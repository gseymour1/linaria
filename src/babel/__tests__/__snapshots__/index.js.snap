// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should extract CSS to a comment 1`] = `
"var background = 'blue';
const Container = component('div', {
  \\"displayName\\": \\"Container\\",
  \\"className\\": \\"Container_1ugh8t9\\",
  \\"interpolations\\": {
    \\"c-0\\": background,
    \\"c-1\\": props => props.color,
    \\"c-2\\": 100 / 3
  }
});

/*CSS OUTPUT START

.Container_1ugh8t9{background-color:var(--c-0);color:var(--c-1);width:var(--c-2);border:1px solid red;}.Container_1ugh8t9:hover{border-color:blue;}

CSS OUTPUT END*/"
`;
